measure_extinction/extdata.py

```diff
@@ -97,81 +98,77 @@ def _get_column_val(column):
+        exts = []
+        for extdata in extdatas:
+            if src in extdata.waves.keys():
+                if src == "BAND":
+                    for i, name in enumerate(extdata.names["BAND"]):
+                        bexts[name].append(extdata.exts["BAND"][i])
+                else:
+                    extdata.exts[src][np.where(extdata.npts[src] == 0)] = np.nan
+                    exts.append(extdata.exts[src])
 
-    # do the photometric band data separately
-    src = "BAND"
-    # possible band waves and number of curves with those waves
-    pwaves = {}
-    pwaves_ext = {}
-    pwaves_uncs = {}
-    pwaves_npts = {}
-    pwaves_names = {}
-    for cext in extdatas:
-        for k, cwave in enumerate(cext.waves[src]):
-            y = cext.exts[src][k]
-            yu = cext.uncs[src][k]
-            if alav is not None:
-                av = _get_column_val(cext.columns["AV"])
-                y = (y / av) + 1.0
-                yu /= av
-            cwavev = cwave.to(u.micron).value
-            cband = cext.names[src][k]
-            if cband in pwaves.keys():
-                pwaves_ext[cband] += y
-                pwaves_uncs[cband] += np.square(yu)
-                pwaves_npts[cband] += 1.0
-            else:
-                pwaves[cband] = cwavev
-                pwaves_ext[cband] = y
-                pwaves_uncs[cband] = np.square(yu)
-                pwaves_npts[cband] = 1.0
-                pwaves_names[cband] = cext.names[src][k]
-
-    aveext.waves[src] = np.array(list(pwaves.values())) * u.micron
-    aveext.exts[src] = np.array(list(pwaves_ext.values())) / np.array(
-        list(pwaves_npts.values())
-    )
-    aveext.uncs[src] = np.sqrt(np.array(list(pwaves_uncs.values()))) / np.array(
-        list(pwaves_npts.values())
-    )
-    aveext.npts[src] = np.array(list(pwaves_npts.values()))
-    aveext.names[src] = np.array(list(pwaves_names))
+        if src == "BAND":
+            aveext.exts["BAND"] = []
+            aveext.npts["BAND"] = []
+            aveext.stds["BAND"] = []
+            aveext.uncs["BAND"] = []
+            for name in aveext.names["BAND"]:
+                aveext.exts["BAND"].append(np.nanmean(bexts[name]))
+                aveext.npts["BAND"].append(len(bexts[name]))
+
+                # calculation of the standard deviation (this is the spread of the sample around the population mean)
+                aveext.stds["BAND"].append(np.nanstd(bexts[name], ddof=1))
+
+                # calculation of the standard error of the average (the standard error of the sample mean is an estimate of how far the sample mean is likely to be from the population mean)
+                aveext.uncs["BAND"].append(stats.sem(bexts[name]))
+
+        else:
+            aveext.exts[src] = np.nanmean(exts, axis=0)
+            aveext.npts[src] = np.sum(~np.isnan(exts), axis=0)
+            aveext.stds[src] = np.nanstd(exts, axis=0, ddof=1)
+            aveext.uncs[src] = stats.sem(exts, axis=0, nan_policy="omit")
```

Comment:
Why not just calculate the uncs from stds directly?  Shouldn't uncs = stds/sqrt(npts)?

Reply:
That is also possible, yes. I used the built-in function because it exists... Do you want me to change it?

Reply:
I changed it.